components/server/src/user/user-service.ts

```diff
@@ -209,6 +216,12 @@ export class UserService {
                     "You're no longer a member of the selected billing team.",
                 );
             }
+            if (await this.isUnbilledTeam(attribution)) {
```

Comment:
👍  This also mitigates https://github.com/gitpod-io/gitpod/issues/13977, but we should still prevent setting it in the first place (server-side).